### PR TITLE
feat(Community Permissions): Integrate In section with permission creation/editing

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Adds tracking of dirty values and storing state for `In` section of community permission.

Closes: #8855 

### Affected areas
`CommunityPermissionsSettingsPanel`, `CommunityNewPermissionView`, `CommunityPermissionsView`

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

[Kazam_screencast_00112.webm](https://user-images.githubusercontent.com/20650004/215483417-611988a5-2236-4794-8ae0-163e4d69e183.webm)

### CODE REVIEW

**Review by noeliaSD — COMMENTED** [review]
Draft reviewed! Code LGTM and after testing, behavior seems the expected one!

Just a minor issue. After creating the permission itself, it is missing the icon in the pills related to **channels** or **community**:

App:
Screenshot 2023-01-27 at 12 06 53

Design:
Screenshot 2023-01-27 at 12 03 16

**Comment by micieslak** [comment]
> Draft reviewed! Code LGTM and after testing, behavior seems the expected one!
> 
> Just a minor issue. After creating the permission itself, it is missing the icon in the pills related to **channels** or **community**:
> 
> App: Screenshot 2023-01-27 at 12 06 53
> 
> Design: Screenshot 2023-01-27 at 12 03 16

Thank you @noeliaSD. I completed the pr by adding what you pointed out.

Screenshot from 2023-01-30 13-15-04

**Comment by noeliaSD** [comment]
> 

Thanks! It looks nice! :)

**Review by noeliaSD — APPROVED** [review]

**Review by alexandraB99 — APPROVED** [review]
looks cool, thanks @micieslak! is this still a draft?

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml [hunk 1]
```diff
@@ -279,15 +337,26 @@ StatusScrollView {
...
+                    ExpressionRole {
+                        name: "text"
+                         // Direct call for singleton function is not handled properly by SortFilterProxyModel that's why `holdingsTextFormat` is used instead.
+                        expression: d.holdingsTextFormat(model.type, model.name, model.amount)
+                    },
+                    ExpressionRole {
+                        name: "operator"
+
+                        // Direct call for singleton enum is not handled properly by SortFilterProxyModel.
+                        readonly property int none: OperatorsUtils.Operators.None
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:350]: Ouch :/

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml [hunk 2]
```diff
@@ -95,20 +99,26 @@ StatusScrollView {
...
-                inModelChannels.clear()
+                //inModelChannels.clear()
+                d.dirtyValues.channelsModel.clear()
                 inSelector.wholeCommunitySelected = true
                 inSelector.itemsModel = inModelCommunity
             } else {
                 inSelector.itemsModel = 0
                 inSelector.wholeCommunitySelected = false
-                inSelector.itemsModel = inModelChannels
+               // inSelector.itemsModel = inModelChannels
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:109]: leftover?

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml [hunk 3]
```diff
@@ -95,20 +99,26 @@ StatusScrollView {
 
         onIsCommunityPermissionChanged: {
             if (isCommunityPermission) {
-                inModelChannels.clear()
+                //inModelChannels.clear()
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:102]: leftover?

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml [hunk 4]
```diff
@@ -95,20 +99,26 @@ StatusScrollView {
...
         }
 
-        property bool triggerDirtyTool: false // Trick: Used to force the reevaluation of dirty when an item of the list is updated
+        // Trick: Used to force the reevaluation of dirty when an item of the list is updated
+        property int triggerDirtyTool: 0
 
         property QtObject dirtyValues: QtObject {
             property ListModel holdingsModel: ListModel {}
+            property ListModel channelsModel: ListModel {}
+            //property var community: null
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:120]: leftover?

File: ui/app/AppLayouts/Chat/stores/CommunitiesStore.qml [hunk 5]
```diff
@@ -91,47 +91,56 @@ QtObject {
...
 
         // Setting PERMISSIONS:
         permission.permissionsObjectModel.key = permissions.key
         permission.permissionsObjectModel.text = permissions.text
         permission.permissionsObjectModel.imageSource = permissions.imageSource
 
+        // Setting CHANNELS
+        for (let c = 0; c < channels.count; c++ ) {
+            const entry = channels.get(c)
+             // roles: type, key, name, amount, imageSource
```
Comment (caybro) [inline ui/app/AppLayouts/Chat/stores/CommunitiesStore.qml:126]: is this comment correct, looks like a copy&paste

**Comment by micieslak** [comment]
> looks cool, thanks @micieslak! is this still a draft?

No, I forgot to mark as ready :)

**Review by glitchminer — COMMENTED** [review]
### QA

Hi @micieslak , please see the notes from tests below:

---
1. OPEN Community color not shown in permissions list
image

---
2. OPEN On editing permission a blank channel is shown in the list
image

---
3. OPEN On editing permissions and changing from Become member to Moderate it's possible to save the permission without selecting any channels. This appears to apply the community value.

https://user-images.githubusercontent.com/50769329/215807636-f55724f6-074c-4fab-bbec-f99b92547298.mp4

**Comment by micieslak** [comment]
Thank you @glitchminer!
 
> 1. OPEN Community color not shown in permissions list

Fixed.

> 2. OPEN On editing permission a blank channel is shown in the list

I cannot reproduce that. Could you provide more details when it happens?

> 3. OPEN On editing permissions and changing from Become member to Moderate it's possible to save the permission without selecting any channels. This appears to apply the community value.

This problem is present also for `Who holds` section. The design is not clear how to handle that situation. I created a separate ticket for that: https://github.com/status-im/status-desktop/issues/9388

**Review by caybro — APPROVED** [review]
LGTM

**Comment by micieslak** [comment]
Hi @glitchminer, I fixed the issue with community color. It should be ok now. Thanks!

**Review by glitchminer — APPROVED** [review]
Thanks for the fixed @micieslak.
Tested, approved.

**Pull request merged**